### DISCUSSION TITLE
How to add an irregular zone and other zone questions

### DISCUSSION THREAD

**Discussion by fokcuk** [rand256/valetudo#444] [Q&A]
Hello 
How would I go about adding an irregular map to a named zone? the interface only offers a rectangular pattern as a single zone
I am also finding that if I send multiple zones through MQTT one after another, it often breaks the map or vacuum thinks its in another location.
So I am currently resorting to do random zones during the week.

**Comment by pidator** [comment]
> How would I go about adding an irregular map to a named zone? the interface only offers a rectangular pattern as a single zone

Zones are only possible in rectangular shape! If you need a "special" shape you'll have to combine two zones. Only forbidden zones can be saved in an irregular shape.

> I am also finding that if I send multiple zones through MQTT one after another, it often breaks the map or vacuum thinks its in another location.

What robot type do you use?

**Reply by fokcuk** [reply]
gen1.
I had an issue with a whole house cleanup, where battery charge was not enough to finish one big zone. It would go back to the dock to charge, but will not resume vacuuming once it's charge. 
Will try doing a whole house by using multiple small zones
